bgpd/bgp_evpn.c

```diff
@@ -6049,6 +6049,8 @@ bool bgp_evpn_is_prefix_nht_supported(const struct prefix *pfx)
 	 */
 	if (pfx && pfx->family == AF_EVPN &&
 	    (evp->prefix.route_type == BGP_EVPN_MAC_IP_ROUTE ||
+	     evp->prefix.route_type == BGP_EVPN_AD_ROUTE ||
```

Comment:
Modification looks straight forward. But are these changes enough ? How/What did you test with this new code ? Could you please share ?

Reply:
This was validated as a part of convergence tests run by Nvidia. While those tests are a part of the Nvidia CI/CD, they do run in all hardware/switch env (non-ssim) and at this point I do not have an easy way to export the convergence tests to the FRR CI/CD.

I noticed that you guys added the nht support for other evpn-types. Is there a test for that which I can extend for type-1?

Reply:
Also I was thinking this function should be just for all evpn routes independent of type so we don't have go through this exercise for every new type that is added. What do you think about that?

Reply:
> Also I was thinking this function should be just for all evpn routes independent of type so we don't have go through this exercise for every new type that is added. What do you think about that?

I think I agree with this. Initially when this function was written, FRR supported multiple EVPN route types, but we added nht support for only type-5 routes and thus this function. But, since all the EVPN route types in FRR now support nht tracking, I think we can make this check independent of route types or get rid of this function and make nht tracking mandatory for all EVPN routes.

Reply:
@AnuradhaKaruppiah Thanks for the info on testing.  @adharkar Can you please answer about topotest ?

Anuradha, It may be better to leave this function as-is. In future, we may support some new route types in EVPN, and not sure if the nht support will be added by default.

Reply:
> @adharkar Can you please answer about topotest ?
> 
>
@srimohans we do not have any specific topotest for this as this was just a bugfix. We have manually tested triggers like:
1. Receive route when the nexthop is unreachable 
2. Receive route when the  the nexthop is reachable.
3. Make nexthop reachable/unreachable for already existing EVPN routes.

Import should happen when the nexthop becomes reachable, unimport should happen when the nexthop becomes unreachable.

Reply:
Thanks for the info guys. We can re-eval usage again when the next evpn route-type is added. 

If there are no other open qs on this commit can you please merge it.